src/components/CodeHighlight/LogColorizer.js

```diff
@@ -0,0 +1,194 @@
+    <span className={css.header__entries}>
+      <FormattedMessage
+        id="ui-data-import.recordsCount"
+        values={{ count: recordsCount }}
+      />
+    </span>
+  );
+
+  const errors = (
+    <span>
+      &#40;
+      <span className={css.header__errors}>
+        <FormattedMessage
+          id="ui-data-import.errorsCount"
+          values={{ count: errorsCount }}
+        />
+      </span>
+      &#41;
+    </span>
+  );
+
+  return (
+    <Fragment>
+      {visible && (
+        <div className={css.toolbar}>
+          <div className={css.header}>{message}&nbsp;{entries}&nbsp;{(errorsCount > 0) && errors}</div>
+          <div className={css.filter}>
+            <span className={css.filter__label}>
+              <FormattedMessage id="ui-data-import.logColorizer.filter.label" />:
+            </span>
+            <ButtonGroup
+              data-test-logs-filter
+              fullWidth
+              role="tablist"
+            >
+              {filterOptions.map(option => (
+                <Button
+                  role="tab"
+                  aria-selected={currentFilter === option.id}
+                  id={`option-${option.id}`}
+                  key={`option-${option.id}`}
+                  buttonStyle={currentFilter === option.id ? 'primary' : 'default'}
+                  className={css.filter__button}
+                  data-test-logs-filter-option={option.id}
+                  marginBottom0
+                  onClick={() => setCurrentFilter(option.id)}
+                >
+                  <FormattedMessage id={option.caption} />
+                </Button>
+              ))}
+            </ButtonGroup>
+          </div>
+          {showThemes && (
+            <div className={css.themes}>
+              <span className={css.themes__label}>
+                <FormattedMessage id="ui-data-import.logColorizer.themes.label" />:
+              </span>
+              <Select
+                onChange={e => setCurrentTheme(e.target.value)}
+                value={currentTheme}
+                marginBottom0
+              >
+                {themesPresent.map(item => (
+                  <option
+                    key={item.id}
+                    value={item.id}
+                  >
+                    {item.caption}
+                  </option>
+                ))}
+              </Select>
+            </div>
+          )}
+        </div>
+      )}
+      <pre
+        id="logs-pane"
+        className={currentTheme}
+      >
+        {codePortion.map((item, i) => {
+          const codeString = JSON.stringify(item, null, 2);
+          const hasError = codeString.includes('error');
+          const showErrors = hasError && OPTIONS.ERRORS === currentFilter;
+          const showInfo = !hasError && OPTIONS.INFO === currentFilter;
+          const showAll = OPTIONS.ALL === currentFilter;
+
+          return (showAll || showErrors || showInfo) && (
+            <CodeHighlight
+              key={`snippet-${i}`}
```

Comment:
An index is not a good choice for key property. item.id would be better.
```suggestion
              key={`snippet-${item.id}`}
```

Reply:
Done

Reply:
Done